### DISCUSSION TITLE
Alternative way to set `connection_limit`

### DISCUSSION THREAD

**Discussion by ChuckJonas** [prisma/prisma#22997] [Q&A]
## Problem

I have a prisma app (dependency via [Keystonejs](https://keystonejs.com/)) configured to run on heroku with a hobby pg instance.  

I'm running into instances where the primsa client is exceeding the connection limit of 20.  Because Heroku sets the `DATABASE_URL` and does not allow you to modify it, I cannot set the [connection_limit](https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/databases-connections/connection-pool#setting-the-connection-pool-size) param on the `DATABASE_URL`.

## Suggested solution

Provide some other ENV or configuration to set `connection_limit`.  Although I'm not sure, I would actually be able to use it in my project as prisma is a third party dependency of keystone.

## Alternatives

Could probably append the env `process.env.DATABASE_URL + '?connection_limit=20'`

**Comment by ludralph** [comment]
Hi @ChuckJonas 👋 

Thank you for raising this question.

Can you please take a look at this section of our documentation on [expanding environment variables](https://www.prisma.io/docs/orm/more/development-environment/environment-variables/env-files#expanding-variables)? This suggestion was also mentioned for a user who experienced similar issue [here](https://github.com/prisma/prisma/issues/11818#issuecomment-1041361447) with Heroku.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ChuckJonas** [reply]
Not sure this will actually help in my case with keystonejs...

I would need a way to tell prisma to now use `DATABASE_URL_WITH_CONNECTION` instead of `DATABASE_URL`.  

The `schema.primsa` file in my project is generated and thus `env("DATABASE_URL")` cannot be overridden...

**Reply by ludralph** [reply]
Hi @ChuckJonas 

In your schema file, you could do something like this:
```
const client = new PrismaClient({ datasources: { db: { url: `${process.env.DATABASE_URL}?connection_limit=10` } } })
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
